File: Client/src/app/source/models/dtos/entities/order-response.ts
import { Grouping } from "../../common/grouping";
import { AddressResponse } from "./address-response";
import { OrderElementResponse } from "./order-element-response";
import { OrderStatusResponse } from "./order-status-response";

export class OrderResponse {
    id!: number;
    deliveryMethod!: string;
    createdAt!: Date;
    updatedAt!: Date;
    comments!: string;
    tipPercent!: number;
    payingWith!: number;
    productTotal!: number;
    productCount!: number;
    shippingCost!: number;
    addressName!: string;
    address!: AddressResponse;
    status!: string;
    orderStatuses!: OrderStatusResponse[];
    orderElements!: OrderElementResponse[];

    _cols: string = "col-md-12";
    _detailsLoaded!: boolean;
    _orderElementsGrouped: Grouping<string, OrderElementResponse>[] = [];
}

File: Server/Source/Models/DTOs/Entities/OrderResponse.cs
using System.Text.Json;

namespace Server.Source.Models.DTOs.Entities
{
    public class OrderResponse
    {
        public int Id { get; set; }
        public string? DeliveryMethod { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public string? Comments { get; set; }
        public decimal? TipPercent { get; set; }
        public decimal? PayingWith { get; set; }
        public decimal? ProductTotal { get; set; }
        public int? ProductCount { get; set; }
        public decimal? ShippingCost { get; set; }
        public string? AddressName { get; set; }
        public string? AddressJson { get; set; }
        public string? Status { get; set; }

        public List<OrderStatusResponse>? OrderStatuses { get; set; }
        public List<OrderElementResponse>? OrderElements { get; set; }
        public AddressResponse? Address { get; set; }
    }
}
